### PR TITLE
Flush HTTP headers and tell Apache that everything is OK

### PR DESCRIPTION
**Pull request by hangy**
**Description:** Try to avoid Apache's built-in error page.
**Related issues and discussion:** #1813

### CODE REVIEW

**Comment by hangy** [comment]
@stephanegigandet This seems to work fine in my Docker container. Can you see if it's the same on dev? 🙂

**Comment by stephanegigandet** [comment]
@hangy : I tried on the dev server but it does not seem to change anything. You do not get the extra header at the bottom in docker?

**Comment by stephanegigandet** [comment]
Maybe it's because of this ModRegistry thing we use. Or the exit() call in display_error... It would be great to remove it and more gracefully exit (which may be contributing to the memory leaks we have)

**Comment by hangy** [comment]
> @hangy : I tried on the dev server but it does not seem to change anything. You do not get the extra header at the bottom in docker?

Not with this change applied ...

> Maybe it's because of this ModRegistry thing we use. Or the exit() call in display_error... It would be great to remove it and more gracefully exit (which may be contributing to the memory leaks we have)

I use the same in Docker https://github.com/openfoodfacts/openfoodfacts-server/blob/7022b7cbe3fa6aad6546f7776efdcc57b7cfa2ca/docker/backend-dev/conf/apache.conf, just because it was used in https://github.com/openfoodfacts/openfoodfacts-server/blob/015154723f19fa5ef0d7b631dc7514738fc6a3c7/conf/apache-2.4/sites-available/off.conf. I literally have no idea about how the CGI process model even works. :slightly_smiling_face:

File: lib/ProductOpener/Display.pm [hunk 1]
```diff
@@ -6256,6 +6256,9 @@ HTML
 	my $status = $request_ref->{status};
 	if (defined $status) {
 		print header ( -status => $status );
```
Comment (stephanegigandet) [inline lib/ProductOpener/Display.pm]: There's already a print header that has been applied in the if elsif else loop above, that seems strange. Am I missing something?

Comment (hangy) [reply lib/ProductOpener/Display.pm]: Those only set cookies or cache policies. The status header is a separate header line, so it should not be (and doesn't appear to be) a problem to sends multiple `print header(...);` calls.

Comment (stephanegigandet) [reply lib/ProductOpener/Display.pm]: It generates an extra empty line, but it seems mod_perl parses the headers anyway, so I guess it's fine.

perl test_header.pl
Expires: Wed, 29 May 2019 09:25:24 GMT
Date: Thu, 30 May 2019 09:25:24 GMT
Content-Type: text/html; charset=UTF-8

Status: 200
Content-Type: text/html; charset=UTF-8

**Comment by stephanegigandet** [comment]
I tested it again, it is working, I must have made a mistake when I tested it earlier on the test server. Thanks @hangy !

**Pull request merged**